walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,121 @@
+  public static final int END_OF_DAY = TimeRange.getTimeInMinutes(23, 59);
+  private static final int TIME_0800AM = TimeRange.getTimeInMinutes(8, 0);
+  private static final int TIME_0830AM = TimeRange.getTimeInMinutes(8, 30);
+  private static final int TIME_0900AM = TimeRange.getTimeInMinutes(9, 0);
+  private static final int TIME_0930AM = TimeRange.getTimeInMinutes(9, 30);
+  private static final int TIME_1000AM = TimeRange.getTimeInMinutes(10, 0);
+  private static final int TIME_1100AM = TimeRange.getTimeInMinutes(11, 00);
+
+  private static final int DURATION_30_MINUTES = 30;
+  private static final int DURATION_60_MINUTES = 60;
+  private static final int DURATION_90_MINUTES = 90;
+  private static final int DURATION_1_HOUR = 60;
+  private static final int DURATION_2_HOUR = 120;
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    //throw new UnsupportedOperationException("TODO: Implement this method.");
+    int counter = 0;
+    int maxHours = TimeRange.WHOLE_DAY.duration();
+    long requestedDuration = request.getDuration();
+    int startTime1 = -1;
+    int endTime1 = -1;
+    int startTime2 = -1;
+    int endTime2 = -1;
+    Collection<String> people = null;
+
+    if(request.getAttendees().isEmpty()){           //if there are no attendees in the request
+        return Arrays.asList(TimeRange.WHOLE_DAY);  //the event can be scheduled at any time of the day
+    }
+    else if(request.getDuration() > maxHours){           //if the duration of the meeting exceeds the amount of time in a day
+        return Arrays.asList();                          //there are no possible meeting times
+    }
+    else if(events.isEmpty()){                      //if there are no scheduled events
+        return Arrays.asList(TimeRange.WHOLE_DAY);  //the event can be scheduled at any time of the day
+    }
+    else if(!events.isEmpty()){
+        for (Event event: events){
+            counter ++;
+            people = event.getAttendees();
+            System.out.println(people);
+            if (counter == 1){                      //on the first event
+                TimeRange time = event.getWhen();   //get the time interval the attendee has a meeting  
+                startTime1 = time.start();          //get the start time of the meeting
+                endTime1 = time.end();              //get the end time of the meeting
+                
+            }
+            else if (counter == 2){
+                TimeRange time = event.getWhen();   //get the time interval the attendee has a meeting  
+                startTime2 = time.start();          //get the start time of the meeting
+                endTime2 = time.end();              //get the end time of the meeting
+            }
+        }
+        //if(!(request.getAttendees() == people)){   //if the people in the request are not the same as the people with events
```

Comment:
That comment is there right now because that piece of code needs work. I needed to submit a pull request but I didn't want to remove it. When it is un-commented, the specific task it was designed for works but a bunch of others don't, therefore I still need to work on it. When I get something working I will remove the dead code if there is any